### ISSUE TITLE
Task 6 - neo4j User ID's 

### ISSUE THREAD

**Issue by animaala** [tomncooper/CSC8101-Documentation#7]
Task 6 asks us to use the demonstration movie db in neo4j. However User nodes don't have a property 'id'. When querying for the ID of a user, or creating a new User node, should we be using/setting the internal id of the node, as there is no 'id' property? (e.g. using the id function).

Thanks

**Comment by hugofirth** [comment]
Hi @animaala 

Firstly, Task 6 does not ask you to use the "demonstration" movie db in neo4j. It asks you to use a much larger version of the same database which can be downloaded [here](https://s3-eu-west-1.amazonaws.com/csc8101-spark-assignment-data/). Please refer to the "The Data" section of the [coursework spec](http://tomcooper.org.uk/CSC8101-Documentation/spark/spark-coursework-spec.html) for more details.

With respect your questions about the ids of `User` nodes, please note that you are expected to create new `User` nodes from user for which you generate ratings. Also note that Neo4j is a schemaless database. The `User` nodes you create do not have to conform (in terms of properties or relationships) to the `User` nodes which already exist. 

In other words, you can set whatever property names you like (though pick sensible ones!) as long as the read queries you write in part 7 still work.

I hope that helps

**Comment by animaala** [comment]
Another small thing (didn't want to make a new issue). When querying the neo4j database, we can explicitly begin and end a transaction from a session, or call run directly on a session and the transaction is implicit. Is there any best practise we should use here. I find calling run on a session personally easier but will I be docked marks for doing so?

**Comment by hugofirth** [comment]
Running directly on a session batches writes into a number of transactions automatically and so should perform well even without your explicitly marking transaction boundaries. You will not be docked marks for relying upon this behaviour.
